modules/promotions/admin-menu-items/go-pro-promotion-item.php

```diff
@@ -3,14 +3,19 @@
 namespace Elementor\Modules\Promotions\AdminMenuItems;
 
 use Elementor\Core\Admin\Menu\Interfaces\Admin_Menu_Item_With_Page;
+use Elementor\Core\Utils\Promotions\Validate_Promotion;
 use Elementor\Settings;
 
 if ( ! defined( 'ABSPATH' ) ) {
 	exit; // Exit if accessed directly
 }
 
 class Go_Pro_Promotion_Item implements Admin_Menu_Item_With_Page {
-	const URL = 'https://go.elementor.com/pro-admin-menu/';
```

Comment:
@Barcov I don't think we can totally remove a const also for BC/3rd party reasons. I suggest keeping the const and changing the logic of `get_url()` to something like this:

```
public static function get_url() {
        $url = self::URL;
        $filtered_url = apply_filters( 'elementor/admin_menu/promotion/upgrade_url', [ 'upgrade_url' => $url ] )['upgrade_url'] ?? '';

        if ( true === Validate_Promotion::domain_is_on_elementor_dot_com( $filtered_url ) ) {
            $url = $filtered_url;
        }

        return esc_url( $url );
    }
```

Reply:
@louiswol94 Interesting suggestion.